packages/models/src/cache/onlineStatus.ts

```diff
@@ -0,0 +1,31 @@
+import { CacheBase } from "./base";
+
+export class OnlineStatus extends CacheBase {
+  readonly key = "online-status";
+
+  async getAll(): Promise<Record<string, string>> {
+    return await this.client.hGetAll(this.key)
```

Comment:
I believe that `hSet` is not the best data stracture for this usecase. I think a list (Array-like) is better for our usecast here. 

- User is online -> Add user id to the list.
- User is offline -> Remove it from the list. 
- Is user online? -> check if user id is in the list.

Reply:
I got your point. I've just found out that `redisClient.get` method has O(1) time complexity [https://redis.io/docs/latest/commands/get/](https://redis.io/docs/latest/commands/get/). However, I think we still shall use `hSet`, not for the purpose of retrieving data specifically, but for checking whether multiple users are online or not within a `multi` - `exec` redis commands, so to speak.

```TypeScript
  async isOnlineBatch(userIds: number[]): Promise<boolean[]> {
    const redisClient = this.client.multi();
    userIds.forEach(id => redisClient.hExists(this.key, this.asField(id)));
    const result = await redisClient.exec();
    return result as unknown as boolean[];
  }
```

Reply:
I got your point. Let's use `set` then.